app/code/Transoft/Blog/view/frontend/web/js/blog-list.js

```diff
@@ -0,0 +1,40 @@
+define([
+    'jquery',
+    'jquery/ui',
+    'underscore',
+    'mage/template'
+], function($, _, mageTemplate){
+    $.widget('mage.blogList', {
+        options: {
+            title: 'title'
+        },
+        /**
+         * Widget initialization
+         * @private
+         */
+        _create: function() {
+            this.renderList();
+        },
+
+        /**
+         * Creates list inside parent container
+         */
+        renderList: function () {
+            let listLength = this.options.length,
+                listData = this.options,
+                titleOutput = mageTemplate.template('<span class="blog-list-title">' +
+                    '<%- title %></span><ul></ul>');
+
+            $(this.element).append(titleOutput({title: this.options.title}));
+
+            for (let i = 0; i < listLength; i++){
+                let listOutput = mageTemplate.template('<li><a href="<%- listData.url %>">' +
+                    '<%= listData.theme %></a></li>');
+
+                $('ul', this.element).append(listOutput({listData: listData[i]}));
+            }
+        }
```

Comment:
Please create underscore template (as separate file), load using define. renderList should look like:
define('text!template-path', tmpl)
renderList: function () {
    var template = _.template(tmpl);
    $(this.element).html(template({items: this.options.items, ....}));
}

Reply:
fixed